src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -678,8 +678,7 @@ type Exception with
 
             let t1, t2, _cxs = NicePrint.minimalStringsOfTwoTypes denv ty1 ty2
 
-            os.Append(ConstraintSolverNullnessWarningEquivWithTypesE().Format t1 t2)
-            |> ignore
+            os.Append(ConstraintSolverNullnessWarningEquivWithTypesE().Format t1) |> ignore
```

Comment:
The change is probably correct, but I would prefer to see at the throw site that indeed the "t1" type is always the non-nullable one and t2 is nullable.

This can be e.g. done by making the pattern match in `SolveNullnessEquiv` total.
Also, the `t2` field is therefore not needed anymore if I got it correctly, the same for `minimalStringsOfTwoTypes ` ?

(= new textual message now displays only the first type and assumes the other ALWAYS is a nullable version of the first)